ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity.component.ts

```diff
@@ -189,52 +187,59 @@ export class ActivityComponent extends BaseActivityComponent implements OnInit,
     public incrementStep(scroll: boolean = true): void {
         const nextIndex = this.nextAvailableSectionIndex();
         if (nextIndex !== -1) {
-            // enable any validation errors to be visible
-            this.validationRequested = true;
-            this.sendSectionAnalytics();
-            this.currentSection.validate();
-
-            if (scroll) {
-                this.scrollToTop();
-            }
-
-            if (this.currentSection.valid) {
-                this.resetValidationState();
-
-                this.isPageBusy.pipe(
-                    tap(_ => this.isLoaded$.next(false)),
-                    debounceTime(this.timeToDebounce),
-                    filter(busy => !busy),
-                    take(1)
-                ).subscribe(_ => {
-                    this.isLoaded$.next(true);
-                    this.currentSectionIndex = nextIndex;
-                    this.visitedSectionIndexes[nextIndex] = true;
-                    this.saveLastVisitedSectionIndex(nextIndex);
-                });
-            }
+            this.submitAnnouncementService.announceSubmit();
+            // The announcement could make listener components busy, but not instantly
+            // introduce a wait before we check whether we busy or not
+            timer(100).pipe(
+                mergeMap(() =>
+                    this.isPageBusy.pipe(
+                        filter(pageIsBusy => !pageIsBusy),
+                        tap(() => {
+                            // run validations and compute flags for blocks to enable scrolling to errors
+                            this.currentSection.validate();
+                            this.model && this.model.shouldScrollToFirstInvalidQuestion();
+                            // triggers scrolling
+                            this.validationRequested = true;
+                        }),
+                        // delay needed for validationRequested to be processed
+                        delay(10),
```

Comment:
I see too many new magic numbers. I'm not that familiar with this piece of functionality as you guys, but I feel like we have problems with chosen approach and can't extend/fix something w/o small delay/timeout hacks with random timers here and where which is not very good... Does any alternatives exist? Or do you know if some refactoring can fix it?

Reply:
In this case, the delay is more of a "yield". In other words, need to allow for asynchronous operations triggered by the assignment:
```typescript
this.validationRequested = true;
```
Otherwise, code in this method would continue to execute and the code that is listening for changes to the validationRequested variable would not execute.

I have updated the delay to be 0. It works just as well:
e994a9f60058af25cadf4afe3c651d19ff574501

I also removed another delay that is not needed in `decrementStep`

Perhaps using a rxjs schedulers would achieve same thing? I have not explored it and don't have a lot of experience with them.

Reality is that my my main concern here is to make sure I can  integrate the mail address component without breaking  existing functionality.

As I have mentioned before, I think this component is overdue for some refactoring. Different people have continued to tack on functionality over years, added their own flags, observables, ngOnChange handlers etcetera to grow it. And as @alextanlan is well aware, we also have to worry about integrating activities inside of activities.

This change to activity.component.ts is not going to be beautiful. Hopefully just enough to work and not introduce bugs.